### PR TITLE
preserve preserveAspectRatio

### PR DESCRIPTION
**Pull request by sathomas**
I suspect that this is a pretty obscure use case, so feel free to ignore/close this pull request. It shouldn't do any harm if you want to merge it, though.

More information on [preserveAspectRatio](http://codepen.io/jonitrythall/post/preserveaspectratio-in-svg).

### CODE REVIEW

**Comment by KittyGiraudel** [comment]
Hey there!

Thanks again for your work. Given that it seems to be an edge case, and the complexity it adds to the original script, I’m going to decline this one. However, if more people are needing it, we’ll reconsider and merge it. :)

What do you think?

**Comment by sathomas** [comment]
No problem. Obviously it's a feature I need, but I don't think too many folks use `preserveAspectRatio` (though it's quite handy).

**Comment by KittyGiraudel** [comment]
Is the sprite broken for you right now?

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 spritesh-*
+.DS_Store
```
Comment (KittyGiraudel) [inline .gitignore:2]: Good point. :D

**Comment by KittyGiraudel** [comment]
I‘m thinking of moving the whole thing to Node.js. With Cheerio and all, would make it easy to do this kind of stuff. Thought?

**Comment by sathomas** [comment]
At the risk of too much information (though perhaps it might be interesting in some weird way), I can elaborate on my specific use case. The icon system consists of color blocks containing the icon imagery. For example:

icon-event

The tricky part is that the size (dimensions and aspect ratio) of the icons vary throughout the site, but I want to preserve the aspect ratio of the imagery. So the above icon might also be rendered as

icon-event

CSS can set the dimensions appropriately, but in order to avoid distorting the imagery, I use `preserveAspectRatio="xMidYMid slice"`.

For reasons that would **definitely** constitute too much information, I can't get away with just having the icon be the outline image and using CSS to set a background color.

So I guess I'd answer the questions as
1. Yes, I do need support for `preserveAspectRatio`, though I'm perfectly happy using my own fork if that support isn't merged. No big deal.
2. I honestly have no opinion either way about switching from bash to node.

**Comment by KittyGiraudel** [comment]
What would be the problem with hard-coding `preserveAspectRatio="xMidYMid slice"` at all time in spritesh?

**Comment by sathomas** [comment]
I don't think I'd recommend fixing `preserveAspectRatio` to `"xMidYMid slice"`. That's not the default value for the attribute, so it might create problems for folks that were expecting default behavior. It would definitely be a breaking change to spritesh.

**Comment by KittyGiraudel** [comment]
The `preserveAspectRatio` will be preserved (see what I did there?) when we merge the Node.js version (#20). :)

**Comment by KittyGiraudel** [comment]
Your problem should be fixed in 1.2.0. :)

**Pull request closed without merging**